### PR TITLE
Fix seeking for buffered input

### PR DESCRIPTION
**Pull request by 4tXJ7f**
CVC4's implementation of seek was calculating the pointer difference
between the current position in the input and the seek point to
determine how many characters to consume. This was causing problems when
ANTLR was seeking to a pointer on a line after the current line because
it would result in a big number of characters to consume because each
line is allocated separately.  This resulted in issue #1113, where CVC4
was computing a large number of characters to consume and would block
until it received all of them.  This commit fixes and simplifies the
code by simply consuming characters until the seek point is reached
without computing a count beforehand.

### CODE REVIEW

**Comment by 4tXJ7f** [comment]
I informed Daniel about the fix. We should wait for his feedback before merging this PR.

**Comment by 4tXJ7f** [comment]
Daniel has confirmed that the fix works for him.

**Comment by ajreynol** [comment]
Unfortunately I have no idea what that code does.  If the fix works for Daniel, it seems fine to me.

File: src/parser/antlr_line_buffered_input.cpp [hunk 1]
```diff
@@ -294,10 +294,8 @@ static void bufferedInputSeek(pANTLR3_INT_STREAM is, ANTLR3_MARKER seekPoint) {
   assert(!line_buffered_input->line_buffer->isPtrBefore(
       (uint8_t*)seekPoint, input->line, input->charPositionInLine));
 
-  ssize_t count = (ssize_t)(seekPoint - (ANTLR3_MARKER)(input->nextChar));
```
Comment (timothy-king) [inline src/parser/antlr_line_buffered_input.cpp]: Looks plausible.

As a sanity check would you do the following for me, add a printf that count how many times the loop / is->consume() is called. Then run this on the example from the issue and respond with the numbers pre & post this commit. If the given explanation is true, you should be able to validate it.

Comment (4tXJ7f) [reply src/parser/antlr_line_buffered_input.cpp]: Unreasonably high counts for seeks were actually how I found the cause of the bug :)

Command line: `cat |  builds/bin/cvc4 --lang=smt2 | nl`

Input (entered line-by-line manually):
```
(set-option :print-success true)
(set-logic ALL)
(define-fun
 __node_init_Store_0
 ((Store.usr.Delta@0 Int))
 Bool
 (let
((X1 0))
   true))
```

Output of fixed version:
```
...
(define-fun
 __node_init_Store_0
 ((Store.usr.Delta@0 Int))
 Bool
 (let
((X1 0))
   true))    17 0
    18  0
    19  0
    20  0
    21  0
    22  0
    23  0
    24  0
    25  0
    26  0
    27  0
    28  0
    29  0
    30  0
    31  0
    32  0
    33  0
    34  4
    35  0
    36  0
    37  0
    38  0
    39  0
    40  0
    41  0
    42  0
    43  0
```

Output of master:
```
(define-fun
 __node_init_Store_0
 ((Store.usr.Delta@0 Int))
 Bool
 (let
((X1 0))
   true))    17 0
    18  0
    19  0
    20  0
    21  0
    22  0
    23  0
    24  0
    25  0
    26  0
    27  0
    28  0
    29  0
    30  0
    31  0
    32  0
    33  0
    34  974
```

**Pull request merged**